src/llvm_backend.cpp

```diff
@@ -1186,38 +1512,308 @@ gb_internal void lb_finalize_objc_names(lbGenerator *gen, lbProcedure *p) {
+                    method_call_args[i+method_param_offset] = lbValue {
+                        wrapper_proc->raw_input_parameters[i+2],
+                        method_type->Proc.params->Tuple.variables[i+method_param_offset]->type,
+                    };
+                }
+                lbValue method_proc_value = lb_find_procedure_value_from_entity(m, md.proc_entity);
+
+                // Call real procedure for method from here, passing the parameters expected, if any.
+                lb_emit_call(wrapper_proc, method_proc_value, method_call_args);
+            }
+            lb_end_procedure_body(wrapper_proc);
+
+
+            // Add the method to the class
+            String method_encoding = str_lit("v");
+            // TODO (harold): Checker must ensure that objc_methods have a single return value or none!
+            GB_ASSERT(method_type->Proc.result_count <= 1);
+            if (method_type->Proc.result_count != 0) {
+                method_encoding = lb_get_objc_type_encoding(method_type->Proc.results->Tuple.variables[0]->type, temporary_allocator());
+            }
+
+            if (!md.ac.objc_is_class_method) {
+                method_encoding = concatenate_strings(temporary_allocator(), method_encoding, str_lit("@:"));
+            } else {
+                method_encoding = concatenate_strings(temporary_allocator(), method_encoding, str_lit("#:"));
+            }
+
+            for (i32 i = method_param_offset; i < method_param_count; i++) {
+                Type *param_type = method_type->Proc.params->Tuple.variables[i]->type;
+                String param_encoding = lb_get_objc_type_encoding(param_type, temporary_allocator());
+
+                method_encoding = concatenate_strings(temporary_allocator(), method_encoding, param_encoding);
+            }
+
+            // Emit method registration
+            lbAddr* sel_address = string_map_get(&m->objc_selectors, md.ac.objc_selector);
+            GB_ASSERT(sel_address);
+            lbValue selector_value = lb_addr_load(p, *sel_address);
+
+            args.count = 4;
+            args[0] = class_value;    // Class
+            args[1] = selector_value; // SEL
+            args[2] = lbValue { wrapper_proc->value, wrapper_proc->type };
+            args[3] = lb_const_value(m, t_cstring, exact_value_string(method_encoding));
+
+            // TODO(harold): Emit check BOOL result and panic if false.
+            lb_emit_runtime_call(p, "class_addMethod", args);
+
+        } // End methods
+
+        // Add ivar if we have one
+        if (ivar_type != nullptr) {
+            // Register a single ivar for this class
+            Type *ivar_base = ivar_type->Named.base;
+
+            const i64 size      = type_size_of(ivar_base);
+            const i64 alignment = type_align_of(ivar_base);
+            // TODO(harold): Checker: Alignment must be compatible with ivar rules. Or we should increase the alignment if needed.
```

Comment:
If an `ivar` needs to have a minimum alignment, I'd enforce it on the declaration, if possible.

Reply:
I revisited Apple's docs: https://developer.apple.com/documentation/objectivec/class_addivar(_:_:_:_:_:)?language=objc

> The instance variable’s minimum alignment in bytes is 1<<align. The minimum alignment of an instance variable depends on the ivar’s type and the machine architecture. For variables of any pointer type, pass log2(sizeof(pointer_type)).

I think in that word salad they meant to say to simply pass the log2 of the alignment, not the actual alignment, which after testing seems to be correct. I validated it by recreating my test case directly in Objective-C and inspecting the ivar offset at runtime.

Fixed now